src/ucs/datastruct/list.h

```diff
@@ -178,7 +184,11 @@ static inline unsigned long ucs_list_length(ucs_list_link_t *head)
     unsigned long length;
     ucs_list_link_t *ptr;
 
-    for (ptr = head->next, length = 0; ptr != head; ptr = ptr->next, ++length);
+    UCS_LIST_CHECK_ELEM_IS_VALID(head);
+    for (ptr = head->next, length = 0; ptr != head; ptr = ptr->next, ++length) {
+        UCS_LIST_CHECK_ELEM_IS_VALID(ptr);
```

Comment:
given the logic above, what is the right thing to check here?

Reply:
as ptr is always being dereferenced, maybe we don't need a check inside the loop